packages/zoe/src/contracts/multipoolAutoswap/swap.js

```diff
@@ -0,0 +1,136 @@
+import { assertProposalKeywords, trade } from '../../contractSupport';
+
+import '../../../exported';
+
+/**
+ *
+ * @param {ContractFacet} zcf
+ * @param {(brand: Brand) => boolean} isSecondary
+ * @param {(brand: Brand) => boolean} isCentral
+ * @param {(brand: Brand) => Pool} getPool
+ */
+export const makeMakeSwapInvitation = (
+  zcf,
+  isSecondary,
+  isCentral,
+  getPool,
+) => {
+  const swapExpected = harden({
+    give: { In: null },
+    want: { Out: null },
+  });
+
+  const swap = seat => {
+    const {
+      give: { In: amountIn },
+      want: { Out: wantedAmountOut },
+    } = seat.getProposal();
+    const { brand: brandIn, value: inputValue } = amountIn;
+    const brandOut = wantedAmountOut.brand;
+
+    // we could be swapping (1) central to secondary, (2) secondary to
+    // central, or (3) secondary to secondary
+
+    if (isCentral(brandIn) && isSecondary(brandOut)) {
+      const pool = getPool(brandOut);
+      const amountOut = pool.getCurrentPrice(true, inputValue);
```

Comment:
when `amountOut` is less than `wantedAmountOut`, should we `exit()` immediately rather than calling `trade()` when we know it should fail?